app/controllers/gift_requests_controller.rb

```diff
@@ -45,7 +45,18 @@ def catalog
     @back_route = events_path
 
     @event = Event.find(params[:id])
-    @gift_requests = @event.gift_requests.unreserved.order(created_at: :asc).page(params[:page] || 1)
+    @gift_requests = @event.gift_requests.unreserved.order(created_at: :asc)
+    if params[:gender] && ["M", "F"].include?(params[:gender])
+      @gift_requests = @gift_requests.where(gender: params[:gender])
+    end
+
+    if params.slice(:min_age, :max_age).any?
+      @gift_requests = @gift_requests.where("age BETWEEN ? AND ?",
+        params[:min_age].present? ? params[:min_age] : 0,
```

Comment:
would be better done as `params[:min_age].presence || 0`, ditto to line below it